File: Java/src/main/java/Solution901_1000/Solution997.java
package Solution901_1000;

/**
 * @author windows
 */
public class Solution997 {
    public int findJudge(int n, int[][] trust) {
        int[] nums = new int[n];
        for (int i = 0; i < trust.length; i++) {
            nums[trust[i][0] - 1] += n;
            nums[trust[i][1] - 1]++;
        }
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == n - 1){
                return i + 1;
            }
        }
        return -1;
    }


    public static void main(String[] args) {
        Solution997 solution997 = new Solution997();
    }
}

File: C++/src/Solution901_1000/Solution997.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution997.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

int findJudge(int n, vector<vector<int>>& trust) {
  vector<int> trusted(n, 0);
  vector<int> beTrusted(n, 0);
  for (const auto &item : trust){
    trusted[item[0] - 1]++;
    beTrusted[item[1] - 1]++;
  }
  vector<int> list;
  for (int i = 0; i < n; i++){
    if (trusted[i] == n - 1 && beTrusted[i] == 0){
        list.push_back(i + 1);
    }
  }
  if (list.size() ==1){
    return list[0];
  }
  return -1;
}
